pkg/protoconv/time.go

```diff
@@ -31,57 +31,44 @@ func ConvertGoGoProtoTimeToGolangProtoTime(gogo *gogoTimestamp.Timestamp) *golan
 }
 
 // ConvertTimestampToTimeOrNow converts a proto timestamp to a golang Time, and returns time.Now() if there is an error.
-func ConvertTimestampToTimeOrNow(gogo *gogoTimestamp.Timestamp) time.Time {
+func ConvertTimestampToTimeOrNow(gogo *timestamppb.Timestamp) time.Time {
 	return ConvertTimestampToTimeOrDefault(gogo, time.Now())
 }
 
 // ConvertTimestampToTimeOrDefault converts a proto timestamp to a golang Time, and returns the default value if there is an error.
-func ConvertTimestampToTimeOrDefault(gogo *gogoTimestamp.Timestamp, defaultVal time.Time) time.Time {
-	t, err := gogoTimestamp.TimestampFromProto(gogo)
-	if err != nil {
+func ConvertTimestampToTimeOrDefault(gogo *timestamppb.Timestamp, defaultVal time.Time) time.Time {
+	if gogo == nil {
 		return defaultVal
 	}
-	return t
+	return gogo.AsTime()
 }
 
 // ConvertTimeToTimestampOrNow converts golang time to proto timestamp.
-func ConvertTimeToTimestampOrNow(goTime *time.Time) *gogoTimestamp.Timestamp {
+func ConvertTimeToTimestampOrNow(goTime *time.Time) *timestamppb.Timestamp {
 	if goTime == nil {
-		return gogoTimestamp.TimestampNow()
+		return timestamppb.Now()
 	}
 	return ConvertTimeToTimestamp(*goTime)
 }
 
 // ConvertTimeToTimestamp converts golang time to proto timestamp.
-func ConvertTimeToTimestamp(goTime time.Time) *gogoTimestamp.Timestamp {
-	t, err := gogoTimestamp.TimestampProto(goTime)
-	if err != nil {
-		return gogoTimestamp.TimestampNow()
-	}
+func ConvertTimeToTimestamp(goTime time.Time) *timestamppb.Timestamp {
+	t := timestamppb.New(goTime)
 	return t
 }
 
 // ConvertTimeToTimestampOrNil converts golang time to proto timestamp or if it fails returns nil.
-func ConvertTimeToTimestampOrNil(goTime time.Time) *gogoTimestamp.Timestamp {
-	t, err := gogoTimestamp.TimestampProto(goTime)
-	if err != nil {
-		log.Error(err)
-		return nil
-	}
-	return t
+func ConvertTimeToTimestampOrNil(goTime time.Time) *timestamppb.Timestamp {
+	return timestamppb.New(goTime)
 }
 
 // MustConvertTimeToTimestamp converts golang time to proto timestamp and panics if it fails.
-func MustConvertTimeToTimestamp(goTime time.Time) *gogoTimestamp.Timestamp {
-	t, err := gogoTimestamp.TimestampProto(goTime)
-	if err != nil {
-		panic(err)
-	}
-	return t
+func MustConvertTimeToTimestamp(goTime time.Time) *timestamppb.Timestamp {
+	return timestamppb.New(goTime)
```

Comment:
What is the difference between these calls now?  Does `timestamppb.New(goTime)` return a nil time or panic if it can't create one?

Reply:
This is exactly what we need! Thank you! I have fixed this and added tests to cover these functions.

The problem is that `gogo - TimestampProto` validates after conversion. And with protobuf V2, we need to do validation explicitly and handle results.

I have added PR for the `master` branch: https://github.com/stackrox/stackrox/pull/12036 - and I'll include that commit here too.